crates/ruma-common/src/events/room/create.rs

```diff
@@ -16,13 +18,16 @@ use crate::{
 /// It acts as the root of all other events.
 #[derive(Clone, Debug, Deserialize, Serialize, EventContent)]
 #[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
-#[ruma_event(type = "m.room.create", kind = State, state_key_type = EmptyStateKey)]
+#[ruma_event(type = "m.room.create", kind = State, state_key_type = EmptyStateKey, custom_redacted)]
 pub struct RoomCreateEventContent {
     /// The `user_id` of the room creator.
     ///
     /// This is set by the homeserver.
-    #[ruma_event(skip_redaction)]
-    pub creator: OwnedUserId,
+    ///
+    /// This is required in room versions 1 trough 10, but is removed starting from room version
+    /// 11.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub creator: Option<OwnedUserId>,
```

Comment:
The spec PR says that clients should always check the sender of the event instead. Let's mark this field as deprecated.

Reply:
It says that for room v11, but auth rules of rooms v1-10 don't change so it's not deprecated until all those room versions are deprecated (if that ever happens).

Reply:
Oh I must have missed that.